// page/userplugin.js











var UserPlugin = function(){
    var self = this,
        container = $(".user-plugin");


    self.init = function(){

        bindEvent();
    }

    function bindEvent(){

        $(document).on("click", "#logout-btn", function(e){
            e.preventDefault();
            $.get("/logout").done(function(data){
                if(data.rcode==200){
                    console.log("container")
                    container.replaceWith(data.data);
                    container = $(".user-plugin");
                }
            })


        })

        // $(document).on("submit", "#user-login form", function(e){
        //     e.preventDefault();
        //     var $this = $(this);
        //     $.post("/login", {
        //         username: $this.find("[name='username']").val(),
        //         password: $this.find("[name='password']").val()
        //     }).done(function(data){
        //         console.log(data);
        //         if(data.rcode==200){
        //             container.replaceWith(data.data);
        //             container = $(".user-plugin");
        //         }
        //     })
        // })

        $(document).on("submit", "#user-register form", function(e){
            e.preventDefault();
            var $this =$(this);
            $.post("/register",{
                username: $this.find("[name='username']").val(),
                password: $this.find("[name='password']").val(),
                nickname: $this.find("[name='nickname']").val(),
            }).done(function(data){
                console.log(data);
                if(data.rcode==200){
                    container.replaceWith(data.data);
                    container = $(".user-plugin");
                }
            })
        })
    }



}






exports.UserPlugin = UserPlugin;
