// model/feed.js

var _ = require("underscore");


var Feed = function(){
    var self = this;

    this.init = function(jsonData){
        this.id = jsonData.id;
        this.title = jsonData.title;
        this.link = jsonData.link;
        this.summary = jsonData.summary;
        this.content = jsonData.content;
        this.createDate = jsonData.createDate;
        this.feedsiteid = jsonData.feedsiteid;
        this.isStared = jsonData.isStared;
        this.isRead = jsonData.isRead;
    }

    this.read = function(callback){
        $.post("/api/feed/" + self.id + "/read", function(data){
            console.log(data);
        })
    }

    this.star = function(callback){
        callback = callback || function(){}
        $.post("/api/feed/" + this.id + "/star").done(function(data){
                    if(data.rcode == 200){
                        callback();
                    }
               })
    }

    this.unstar = function(callback){
        callback = callback || function(){}
        $.post("/api/feed/" + this.id + "/unstar").done(function(data){
                if(data.rcode == 200){
                    callback();
                }
           })
    }

    return this;

}


var FeedList = function(){

    var feedList = []; // 
    this.length = 0;


    this.addFeed = function(feed){
        feedList.push(feed);
        this.length ++;
    }

    this.getFeedById =function(feedid){
        return _.find(feedList, function(x){ return x.id===feedid });
    }

    // this.readAllFeeds = function(){
    //     // make a feedsite api?

    //     _.each(feedList, function(feed, index, list){
    //         feed.read();
    //     })
    // }

    this.render = function(tpl){
        return tpl(feeds=feedList);
    }

    return this;

}



exports.Feed = Feed;
exports.FeedList = FeedList;
